packages/components/socialFeed/Map/Map.web.tsx

```diff
@@ -0,0 +1,254 @@
+import "./styles.css";
+import "leaflet/dist/leaflet.css";
+import { DivIcon, LatLngBounds, point, PointExpression } from "leaflet";
+import { FC, useEffect, useMemo, useState } from "react";
+import {
+  MapContainer,
+  TileLayer,
+  Marker,
+  Popup,
+  useMapEvents,
+  useMap,
+} from "react-leaflet";
+import MarkerClusterGroup from "react-leaflet-cluster";
+import { HeatmapLayer } from "react-leaflet-heatmap-layer-v3/lib";
+import { View } from "react-native";
+
+import { Post } from "@/api/feed/v1/feed";
+import { MapProps } from "@/components/socialFeed/Map/Map.types";
+import { ArticleMapPost } from "@/components/socialFeed/Map/MapPosts/ArticleMapPost";
+import { MusicMapPost } from "@/components/socialFeed/Map/MapPosts/MusicMapPost";
+import { NormalMapPost } from "@/components/socialFeed/Map/MapPosts/NormalMapPost";
+import { PictureMapPost } from "@/components/socialFeed/Map/MapPosts/PictureMapPost";
+import { VideoMapPost } from "@/components/socialFeed/Map/MapPosts/VideoMapPost";
+import { useFetchFeedLocation } from "@/hooks/feed/useFetchFeed";
+import { useSelectedNetworkId } from "@/hooks/useSelectedNetwork";
+import {
+  DEFAULT_MAP_POSITION,
+  getMapPostIconColorRgba,
+  getMapPostIconSVGString,
+  MAP_LAYER_URL,
+} from "@/utils/feed/map";
+import { zodTryParseJSON } from "@/utils/sanitize";
+import {
+  CustomLatLngExpression,
+  PostCategory,
+  ZodSocialFeedPostMetadata,
+} from "@/utils/types/feed";
+
+interface MarkerPopup {
+  position: CustomLatLngExpression;
+  post: Post;
+  fileURL?: string;
+}
+
+// Custom map post icon
+const postIcon = (postCategory: PostCategory) => {
```

Comment:
should memo

Reply:
I think we should follow the example given by react-leaflet-cluster: https://codesandbox.io/p/sandbox/beautiful-pike-j2l0w?file=%2Fsrc%2FApp.tsx

https://github.com/akursat/react-leaflet-cluster

Reply:
Why memo ? We don't want to rerender the whole `Map` component depending on `postIcon`

Reply:
Also, I don't want to rerender `postIcon`, I won't to put it into the `Map` component